pkg/controller/nodemaintenance/check_lease_update.go

```diff
@@ -0,0 +1,481 @@
+}
+
+func compareTime(rval *metav1.MicroTime, lval *metav1.MicroTime) bool {
+	return (rval == nil && lval == nil) || (rval != nil && lval != nil && (*rval).Time == (*lval).Time)
+}
+
+func compareInt32(rval *int32, lval *int32) bool {
+	return (rval == nil && lval == nil) || (rval != nil && lval != nil && *rval == *lval)
+}
+
+func makeString(val string) (*string) {
+	return &val
+}
+
+
+func getTimeLease(tmsec int64) (*metav1.MicroTime) {
+	tm  := time.Unix(int64(tmsec),0)
+	ret := metav1.MicroTime{Time: tm}
+	return &ret
+}
+
+func getMockNode() (*corev1.Node) {
+	node := &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "miau",
+					UID:  "foobar",
+				},
+	}
+	return node
+}
+
+var _ = Describe("checkLeaseUpdate", func() {
+
+	DescribeTable("check lease supported",
+		func(initLease *coordv1beta1.Lease, leasePostcondition *coordv1beta1.Lease, expectedError error) {
+			node := getMockNode()
+			objs := []runtime.Object{
+				initLease,
+			}
+			cl  := fake.NewFakeClient(objs...)
+			timeNow :=  time.Unix(NowTime,0)
+
+			name := apitypes.NamespacedName{Namespace: LeaseNamespace, Name: node.Name}
+			leaseIn := &coordv1beta1.Lease{}
+			err := cl.Get(context.TODO(), name, leaseIn)
+			Expect(err).NotTo(HaveOccurred())
+
+			leaseRet, err, failedUpdateOwnedLease := updateLease(cl, node, leaseIn, timeNow, LeaseDurationInSeconds)
+
+			if leasePostcondition == nil {
+				Expect(err).To(HaveOccurred())
+				Expect(expectedError).To(Equal(err))
+
+				var nilRef *coordv1beta1.Lease
+				Expect(leaseRet).To(Equal(nilRef))
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(expectedError).NotTo(HaveOccurred())
+
+				Expect(failedUpdateOwnedLease).To(Equal(false))
+				lease := &coordv1beta1.Lease{}
+				err = cl.Get(context.TODO(), name, lease)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(leaseRet).To(Equal(lease))
+
+				Expect(len(lease.ObjectMeta.OwnerReferences)).To(Equal(1))
+				Expect(len(leasePostcondition.ObjectMeta.OwnerReferences)).To(Equal(1))
+
+				leaseRef := lease.ObjectMeta.OwnerReferences[0]
+				leasePost := leasePostcondition.ObjectMeta.OwnerReferences[0]
+
+				Expect(leasePost.APIVersion).To(Equal(leaseRef.APIVersion))
+				Expect(leasePost.Kind).To(Equal(leaseRef.Kind))
+				Expect(leasePost.Name).To(Equal(leaseRef.Name))
+				Expect(leasePost.UID).To(Equal(leaseRef.UID))
+
+				Expect(compareString(lease.Spec.HolderIdentity, leasePostcondition.Spec.HolderIdentity)).To(Equal(true))
+				Expect(compareTime(lease.Spec.RenewTime, leasePostcondition.Spec.RenewTime)).To(Equal(true))
+				Expect(compareTime(lease.Spec.AcquireTime, leasePostcondition.Spec.AcquireTime)).To(Equal(true))
+				Expect(compareInt32(lease.Spec.LeaseDurationSeconds, leasePostcondition.Spec.LeaseDurationSeconds)).To(Equal(true))
+
+				Expect(compareInt32(lease.Spec.LeaseTransitions, leasePostcondition.Spec.LeaseTransitions)).To(Equal(true))
```

Comment:
yes, but that makes it harder to figure out what failed.

Reply:
Well, the spec isn't that big, easy to spot the difference.
However, if you prefer to keep the single expectations, please reuse `func ExpectEqualWithNil()` (works as it is) instead of all the `compareXxx` funcs.

Reply:
ping